### DISCUSSION TITLE
Making a README in Kvrocks purely for developers

### DISCUSSION THREAD

**Discussion by jonathanc-n** [apache/kvrocks#2551] [General]
To make it easier for new developers to get started on Kvrocks, I'm proposing to add a README in the repository (for a developer's convenience):
- Explain the certain folders that hold certain functionality (ex. search folder -> KQIR)
- Describe certain functionality (ex. MakeCmdAttr, metadata handling, etc.)
- Committing guidelines, such as splitting larger PRs into smaller commits making it easier for reviewers. (referenced from Meta's Velox)

These were just a few ideas that I came up with off the top of my head. If you can think of more, please feel free to add!

**Comment by tisonkun** [comment]
This is typically called "CONTRIBUTING.md" :D

**Reply by jonathanc-n** [reply]
Oh yeah lmfao, I forgot
